lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+        self.fg_cube.data[0, 7, 7] = 0.0
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 1.0
+        result = plugin.apply_precip(self.fg_cube, self.precip_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_precip_intense_null(self):
+        """Test that low prob of intense precip does not increase
+        lightning risk"""
+        self.precip_cube.data[0, 0, 7, 7] = 1.0
+        self.precip_cube.data[1, 0, 7, 7] = 1.0
+        self.precip_cube.data[2, 0, 7, 7] = 0.1
+        # Set first-guess to zero
+        self.fg_cube.data[0, 7, 7] = 0.1
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.25  # Heavy-precip result only
+        result = plugin.apply_precip(self.fg_cube, self.precip_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+
+class Test_apply_ice(IrisTest):
+
+    """Test the apply_ice method."""
+
+    def setUp(self):
+        """Create cubes with a single zero prob(precip) point.
+        The cubes look like this:
+     precipitation_amount / (kg m^-2)
+     Dimension coordinates:
+        time: 1;
+        projection_y_coordinate: 16;
+        projection_x_coordinate: 16;
+     Auxiliary coordinates:
+          forecast_period (on time coord): 0.0 hours (simulates nowcast data)
+     Scalar coordinates:
+          forecast_reference_time: 2015-11-23 03:00:00
+     Data:
+       self.fg_cube:
+          All points contain float(1.)
+          Cube name is "probability_of_lightning".
+       self.ice_cube:
+          With extra coordinate of length(3) "threshold" containing
+          points [0.5, 1., 2.] kg m^-2.
+          Time and forecast_period dimensions "sqeezed" to be Scalar coords.
+          All points contain float(0.)
+          Cube name is "probability_of_vertical_integral_of_ice".
+"""
+        self.fg_cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube_with_no_realizations(zero_point_indices=[]),
+            fp_point=0.0)
+        self.fg_cube.rename("probability_of_lightning")
+        self.ice_cube = squeeze(
+            add_forecast_reference_time_and_forecast_period(
+                set_up_cube(num_realization_points=3,
+                            zero_point_indices=[]),
+                fp_point=0.0))
+        threshold_coord = self.ice_cube.coord('realization')
+        threshold_coord.points = [0.5, 1.0, 2.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('kg m^-2')
+        self.ice_cube.data = np.zeros_like(self.ice_cube.data)
+        self.ice_cube.rename("probability_of_vertical_integral_of_ice")
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
```

Comment:
Is it worth having a `self.plugin` for `Test_apply_precip` and `Test_apply_ice`?

Reply:
I think so. They need to have had the __init__ funciton executed and it is consistent throughout (I think).